transaction/builder/src/transaction_builder.rs

```diff
@@ -907,11 +837,12 @@ impl<FPR: FogPubkeyResolver> TransactionBuilder<FPR> {
     >(
         self,
         ring_signer: &S,
+        memo_builder: impl MemoBuilder,
         rng: &mut RNG,
     ) -> Result<Tx, TxBuilderError> {
         let fee_map = self.fee_map.clone();
-        let unsigned_tx = self.build_unsigned::<O>()?;
-        Ok(unsigned_tx.sign(ring_signer, fee_map.as_ref(), rng)?)
+        self.build_blueprint()?
+            .sign::<O, S, RNG>(ring_signer, memo_builder, rng, fee_map.as_ref())
```

Comment:
I think this should be closer to the previous implementation.

The reason being that this will ensure that the `build_unsigned()` is used similarly for consumers that build and then later sign
As it stands now, it's possible that this implementation and the `build_unsigned` above deviate from each other.
```suggestion
        let unsigned_tx = self.build_unsigned::<O>(memo_builder)?;
        Ok(unsigned_tx.sign(ring_signer, fee_map.as_ref(), rng)?)
```